app/src/main/java/com/douglasbruce/tasky/features/login/LoginViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.douglasbruce.tasky.features.login
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.douglasbruce.tasky.core.domain.validation.EmailValidator
+import com.douglasbruce.tasky.features.login.form.LoginFormEvent
+import com.douglasbruce.tasky.features.login.form.LoginFormState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+const val loginStateKey = "loginState"
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val savedStateHandle: SavedStateHandle,
+    private val emailValidator: EmailValidator,
+) : ViewModel() {
+
+    var state by mutableStateOf(
+        savedStateHandle[loginStateKey] ?: LoginFormState()
+    )
+        private set
```

Comment:
There should also be an API for directly getting a Compose state instance from a `SavedStateHandle`:
```
var state by savedStateHandle.saveable {
    mutableStateOf(LoginFormState())
}
    private set
```
Then, you can update the state just as usual, but thanks to the delegate it updates `savedStateHandle`.

You might need a separate dependency for that though, not sure about that right now

Reply:
Does this mean I can remove the `updateState` method? and update the state with `state = state.copy()`?

Reply:
Yes, the delegate will make sure that savedStateHandle is updated behind the scenes